plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -153,13 +166,16 @@
 import static java.util.Objects.requireNonNull;
 import static java.util.function.Function.identity;
 import static java.util.stream.Collectors.joining;
+import static org.apache.iceberg.ReachableFileUtil.metadataFileLocations;
+import static org.apache.iceberg.ReachableFileUtil.versionHintLocation;
+import static org.apache.iceberg.TableProperties.WRITE_LOCATION_PROVIDER_IMPL;
 
 public class IcebergMetadata
         implements ConnectorMetadata
 {
     private static final Logger log = Logger.get(IcebergMetadata.class);
-
     private static final Pattern PATH_PATTERN = Pattern.compile("(.*)/[^/]+");
+    private static final int VACUUM_MAX_SUPPORTED_TABLE_VERSION = 2;
```

Comment:
Do we support version 2, or just 1? 

v2 tables can have delete marker files which need to be preserved, do we have a test with those?

Reply:
Those delete marker files are linked from snapshots manifests so they are taken into account. I do have a test `testVacuumingPartitionedTable` that verifies that DeleteFiles are correctly removed.